airbyte-integrations/bases/connector-acceptance-test/unit_tests/test_core.py

```diff
@@ -277,30 +278,94 @@ def test_configured_catalog_fixture(mocker, test_strictness_level, configured_ca
+            {},
+            does_not_raise()
+        ),
         (
             {"type": "object", "properties": {"created": {"type": "string"}}},
+            {},
+            ExpectedRecordsConfig(path="foobar"),
             {"aa": 23},
+            {},
             pytest.raises(AssertionError, match="should have some fields mentioned by json schema"),
         ),
-        ({"type": "object", "properties": {"created": {"type": "string"}}}, {"created": "23"}, does_not_raise()),
         (
             {"type": "object", "properties": {"created": {"type": "string"}}},
+            {},
+            ExpectedRecordsConfig(path="foobar"),
+            {"created": "23"},
+            {},
+            does_not_raise()
+        ),
+        (
+            {"type": "object", "properties": {"created": {"type": "string"}}},
+            {},
+            ExpectedRecordsConfig(path="foobar"),
             {"root": {"created": "23"}},
+            {},
             pytest.raises(AssertionError, match="should have some fields mentioned by json schema"),
         ),
         # Recharge shop stream case
         (
             {"type": "object", "properties": {"shop": {"type": ["null", "object"]}, "store": {"type": ["null", "object"]}}},
+            {},
+            ExpectedRecordsConfig(path="foobar"),
             {"shop": {"a": "23"}, "store": {"b": "23"}},
+            {},
+            does_not_raise(),
+        ),
+        # Fail when expected and actual records are not equal
+        (
+            {"type": "object"},
+            {},
+            ExpectedRecordsConfig(path="foobar"),
+            {"constant_field": "must equal", "fast_changing_field": [{"field": 2}]},
+            {"test_stream": [{"constant_field": "must equal", "fast_changing_field": [{"field": 1}]}]},
+            pytest.raises(Failed, match="Stream test_stream: All expected records must be produced"),
+        ),
+        # Expected and Actual records are not equal but we ignore fast changing field
+        (
+            {"type": "object"},
+            {"test_stream": [IgnoredFieldsConfiguration(name="fast_changing_field/*/field", bypass_reason="test")]},
+            ExpectedRecordsConfig(path="foobar"),
+            {"constant_field": "must equal", "fast_changing_field": [{"field": 2}]},
+            {"test_stream": [{"constant_field": "must equal", "fast_changing_field": [{"field": 1}]}]},
+            does_not_raise(),
+        ),
+        # Fail when expected and actual records are not equal and exact_order=True
+        (
+            {"type": "object"},
+            {},
+            ExpectedRecordsConfig(extra_fields=False, exact_order=True, extra_records=True, path="foobar"),
+            {"constant_field": "must equal", "fast_changing_field": [{"field": 2}]},
+            {"test_stream": [{"constant_field": "must equal", "fast_changing_field": [{"field": 1}]}]},
+            pytest.raises(AssertionError, match="Stream test_stream: Mismatch of record order or values"),
```

Comment:
I'm not sure if I understand why this fails for the exact order one. Shouldn't it fail because the values don't match? An exact order test would be if we had multiple records output for the stream, which match the initial ones but are out of order

Reply:
I need this test to ensure that my code works in `if exact_order:` branch

Reply:
Got it - I misinterpreted this as a new test for the exact_order behavior